File: MSP_Backend/src/main/java/de/szut/msp_backend/models/item/GenericItem.java
package de.szut.msp_backend.models.item;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;


// @Data gibt uns alle Getter und Setter ohne sie ausschreiben zu müssen
@Data
@NoArgsConstructor
@EqualsAndHashCode
public abstract class GenericItem
{
    protected int itemID;
    protected String displayName;
    protected String description;
    protected int standardValue;
    protected ItemType itemType;
    protected Rarity rarity;

    /**
     * Creates a generic item.
     *
     * @param itemID        unique ID of the Item.
     * @param displayName   that the player sees while playing.
     * @param description   for the player, to know what this item can do.
     * @param standardValue price if the player wants to sell the item or buy it.
     * @param itemType      consumable, weapon or tradeItem
     * @param rarity        plays part in how often the item will appear in the game.
     */
    protected GenericItem(int itemID, String displayName, String description, int standardValue, ItemType itemType, Rarity rarity)
    {
        this.itemID = itemID;
        this.displayName = displayName;
        this.description = description;
        this.standardValue = standardValue;
        this.itemType = itemType;
        this.rarity = rarity;
    }

    @Override
    public String toString()
    {
        return "{" + "\"itemID\":" + itemID + ", \"displayName\":\"" + displayName + "\", \"description\":\"" + description + "\", \"standardValue\":" + standardValue + ", \"itemType\":\"" + itemType + "\", \"rarity\":\"" + rarity + "}";
    }
}

File: msp_frontend/src/models/items/GenericItem.ts
import { ItemType } from "../../enums/ItemType"
import { Rarity } from "../../enums/Rarity"

/**
 * @interface interface representing a generic item.
 */
export interface GenericItem {
  /**
   * @description The item's internal id.
   */
  itemID: number;
  /**
   * @description The item's name.
   */
  displayName: string;
  /**
   * @description Description of the item.
   */
  description: string;
  /**
   * @description the default buy/sell value of the item.
   */
  standardValue: number;
  /**
   * @description the item's type. In this case a CONSUMABLE.
   */
  itemType: ItemType;
  /**
   * @description The item's rarity.
   */
  rarity: Rarity;
}
